mteb/abstasks/AbsTaskCLSD.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+from time import time
+
+from ..evaluation.evaluators import RetrievalEvaluator
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+
+class AbsTaskCLSD(AbsTask):
+    """Abstract class for Cross Lingual Semantic Discrimination Tasks. Initialised from starting code of AbsTaskRetrieval
```

Comment:
Please add a line describing the task as well

Reply:
Any reason to not just use a retrieval task?

Reply:
Thanks for outlining it @Andrian0s, sounds like it is compatible. If that is the case I would much prefer if we keep it the same. This makes maintaining the code notably simpler going forward and since the change is only an added metric to the retrieval task I don't believe it introduced any issues.

CLSD can then become a task subtype.

I will just get a second opinion from @orionw here as well (he did the AbsTaskInstructionRetrieval)

Reply:
Thanks for the great work @Andrian0s! 

We extended `AbsTaskInstructionRetrieval` primarily because it evaluates by comparing the difference between two runs as opposed to the standard one run in a retrieval setup. Although there was a different metric, it was more about the fact that the metric needs the model to be run twice over the data and compare rather than the new metric itself -- and no previous abstract task let us run it twice.

You make good points about the pros to separating, but I think the extra time of calculating more metrics is negligible - it should just take at most a few more seconds to calculate the standard metrics beyond accuracy.

--------------

I do think you are right that the [current setup of the dataset](https://huggingface.co/datasets/Andrianos/clsd_wmt19_21/tree/main/wmt19.fr.de) is not in any of the current AbstractTask formats: you have the source, target, and the adversarial sentence as one row in the dataset.

Looking at the task description, I could see this being a retrieval task (a cross-lingual "find the correct sentence among the collection which includes distractors") if you combine all the non-source sentences into one collection. As long as no other sentences are closer to the source than the target and the adversarial examples (which I think is a correct assumption), this should end up the same as your current evaluation setup at recall@1.

On the other hand, your task is also very similar to a classification task, where you want to predict True for the target sentence and False for the adversarial sentences.  This might be truer to your current evaluation if you really only care about accuracy and you're okay with averaging across all instances (whereas retrieval is averaged per source sentence).

However, a third setup that might also work well (although I'm less familiar with it) is that it could probably be formatted as bitext mining. In that case you're looking to mine the correct sentence and not the others - though I think @KennethEnevoldsen is more familiar with this task than me and can weigh in on it.

You are the most familiar with your dataset, but it seems like this task could fit into one of these three topics -- unless I'm misunderstanding something

Reply:
Thank you @KennethEnevoldsen  and @orionw for the quick suggestions, promptness really helps.

I appreciate the discussion and I followed Orion's suggested and instead used the AbsRetrieval class. A lot of the evaluations are irrelevant (but not causing an issue), but since we will only use recall_at_1 as the main (which is same as our accuracy, verified), no harm done. 

To be fair, recall_at_3 isn't such a bad insight either. 

I think it's good to go, current implementation is simplistic and sufficient.

Reply:
Thanks for the changes and the considerations here - For now the tab is not required in MTEB, but can be added later